### ISSUE TITLE
notifyError does not track and send NSError.userInfo

### ISSUE THREAD

**Issue by bolsinga** [bugsnag/bugsnag-cocoa#544]
### Description
We used NSError.userInfo to send along additional information for other crash reporters "soft error logging". We use +[Bugsnag notifyError:block:] for this purpose. The NSError.userInfo does not appear in these bugsnag logs.

### Issue

In the following code in BugsnagNotifier.m, the NSError.userInfo is used to create the NSException wrapper, but the error.userInfo is not added to the meta data. It doesn't seem to be pulled from the NSException wrapper either. This would be useful information to have up on the web interface.

```
- (void)notifyError:(NSError *)error
              block:(void (^)(BugsnagCrashReport *))block {
    BugsnagHandledState *state =
        [BugsnagHandledState handledStateWithSeverityReason:HandledError
                                                   severity:BSGSeverityWarning
                                                  attrValue:error.domain];
    NSException *wrapper = [NSException exceptionWithName:NSStringFromClass([error class])
                                                   reason:error.localizedDescription
                                                 userInfo:error.userInfo];
    [self notify:wrapper
        handledState:state
               block:^(BugsnagCrashReport *_Nonnull report) {
                 NSMutableDictionary *metadata = [report.metaData mutableCopy];
                 metadata[@"nserror"] = @{
                     @"code" : @(error.code),
                     @"domain" : error.domain,
**// FIXME add userInfo here.**
                     BSGKeyReason : error.localizedFailureReason ?: @""
                 };
                   if (report.context == nil) { // set context as error domain
                       report.context = [NSString stringWithFormat:@"%@ (%ld)", error.domain, (long)error.code];
                   }
                 report.metaData = metadata;

                 if (block) {
                     block(report);
                 }
               }];
}


```### Environment

Library versions:

* Bugsnag version: master

</details>

**Comment by mattdyoung** [comment]
Hi @bolsinga 

What is it specifically within the `NSError.userInfo` that you'd like to see on the user interface which isn't currently displayed?

The most important keys within `userInfo` are already carried through i.e. `NSLocalizedDescriptionKey` shows up as the main error description and `NSLocalizedFailureReasonErrorKey` shows up on the `NSError` tab as `reason`.

**Comment by bolsinga** [comment]
I think those from from the `NSError` as properties. `NSError.userInfo` is another key/value dictionary property on `NSError`.

We put meta data for our particular errors in there for logging to view later on our bug reports.

Thanks!

**Comment by mattdyoung** [comment]
Those are keys in `userInfo` but they also get returned as properties on the `NSError` as well if set:
https://developer.apple.com/documentation/foundation/userinfokey

It seems a reasonable feature request for us to include the other keys as well. We'll look into doing this when priorities allow.

**Comment by bolsinga** [comment]
Thanks! If I have a chance I’ll submit a PR.

**Comment by jgreen210** [comment]
If use NSLocalizedFailureReasonErrorKey in NSError userInfo, then that information appears in the bugsnag console, but the grouping of crashes is affected too, which is generally not what want (it can easily create far too many groups of issues in the bugsnag console). It would be useful if other userInfo fields are reported too, but that the console's grouping is not affected.

This is for the bugsnag-cocoa version used by BugsnagReactNative (2.23.2) and https://docs.bugsnag.com/platforms/ios/#reporting-handled-exceptions ios API.

**Comment by mattdyoung** [comment]
Hi @jgreen210 

Whilst the original request here is something we plan to work on when priorities allow, using `NSLocalizedFailureReasonErrorKey` shouldn't affect error grouping. Events would typically be grouped by the file, line number and error class (i.e. `NSError`) unless the default grouping has been overridden. Perhaps setting the `NSLocalizedFailureReasonErrorKey` is also shifting the line numbers so it's the changed line number that's affecting the grouping?

If that doesn't make sense of what you're seeing I'd suggest writing in to Bugsnag support with some links to examples of incorrect grouping in your dashboard and we can investigate.

**Comment by jgreen210** [comment]
The line number probably did change, so I've given my comment a thumbs down.

**Comment by yousif-bugsnag** [comment]
This has been released in [v6.6.0](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.6.0).
